packages/blocks/src/api/registration.js

```diff
@@ -310,6 +316,115 @@ export function registerBlockType( name, settings ) {
 	return settings;
 }
 
+/**
+ * Translates block settings provided with metadata using the i18n schema.
+ *
+ * @param {string|string[]|Object[]} i18nSchema    I18n schema for the block setting.
+ * @param {string|string[]|Object[]} settingValue  Value for the block setting.
+ * @param {string}                   textdomain    Textdomain to use with translations.
+ *
+ * @return {string|string[]|Object[]} Translated setting.
+ */
+function translateBlockSettingUsingI18nSchema(
+	i18nSchema,
+	settingValue,
+	textdomain
+) {
+	if ( isString( i18nSchema ) && isString( settingValue ) ) {
+		// eslint-disable-next-line @wordpress/i18n-no-variables, @wordpress/i18n-text-domain
+		return _x( settingValue, i18nSchema, textdomain );
```

Comment:
In the branch of PR https://github.com/WordPress/gutenberg/pull/31120, I made a simple test and replaced "return _x( settingValue, i18nSchema, textdomain );" with "return 'ok';". Nothing changed. With this change shouldn't all the block titles and descriptions appear as "ok"?

Reply:
It might require adding a fallback to read the value from the client when it isn't provided from the server. It would be useful for WordPress 5.7 when using with the Gutenberg plugin.

Reply:
I investigate it further and it's a different use case than `apiVersion` setting that I was referring to:

https://github.com/WordPress/gutenberg/blob/c047e2716149c794eebff3c2c002f66a6f546f59/packages/blocks/src/api/registration.js#L167-L181

We register all blocks on the server and we set now all translatable fields there and they are exposed to the client. The values from the server take precedence to respect all PHP filters that could get applied to the metadata. It's the same data after all, so there is no reason to override it with the same metadata read on the client. 

To test it, you could either tweak `register_block_type_from_metadata` calls on the server to ignore translatable fields using [`block_type_metadata_settings`](https://developer.wordpress.org/reference/hooks/block_type_metadata_settings/) filter or change the method I embedded above and enforce that those fields get overridden with the values loaded on the client.

Reply:
Thank you for the clarification 👍